var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -60,25 +75,41 @@ def torf(s):
             "Returns the string TRUE or FALSE"
             return repr(s in spec).upper()
 
-        makefile = FileFilter('GNUmakefile')
-        makefile.filter('AMREX_HOME .*', 'AMREX_HOME = amrex')
-        makefile.filter('PICSAR_HOME .*', 'PICSAR_HOME = picsar')
-        makefile.filter('COMP .*', 'COMP = {0}'.format(comp))
-        makefile.filter('DIM .*',
-                        'DIM = {0}'.format(int(spec.variants['dims'].value)))
-        makefile.filter('USE_PSATD .*',
-                        'USE_PSATD = {0}'.format(torf('+psatd')))
+        use_mpi = torf('+mpi')
         try:
             self.compiler.openmp_flag
         except UnsupportedCompilerFlag:
             use_omp = 'FALSE'
         else:
-            use_omp = torf('+openmp')
+            use_omp = torf('backend=openmp')
+        use_cuda = torf('backend=cuda')
+        use_hip = torf('backend=hip')
         use_openpmd = torf('+openpmd')
+        use_ascent = torf('+ascent')
+
```

Comment:
There's a small inconsistency. For some of the checks, there are local variables, e.g. use_cuda. But for others, the call to torf is from inside makefile.filter, e.g. for psatd. My 2cents is I think it's more readable, simpler without the extra local variable (except when needed as with use_omp).

Reply:
Thanks, yes `torf('+psatd')` is the only one that is not defining a variable first for readability.

Reply:
simplified now :)